src/font/sprite/Face.zig

```diff
@@ -16,25 +16,154 @@ const std = @import("std");
 const builtin = @import("builtin");
 const assert = std.debug.assert;
 const Allocator = std.mem.Allocator;
+const wuffs = @import("wuffs");
+const z2d = @import("z2d");
 const font = @import("../main.zig");
 const Sprite = font.sprite.Sprite;
-const Box = @import("Box.zig");
-const Powerline = @import("Powerline.zig");
-const underline = @import("underline.zig");
-const cursor = @import("cursor.zig");
+
+const special = @import("draw/special.zig");
 
 const log = std.log.scoped(.font_sprite);
 
 /// Grid metrics for rendering sprites.
 metrics: font.Metrics,
 
+pub const DrawFnError =
+    Allocator.Error ||
+    z2d.painter.FillError ||
+    z2d.painter.StrokeError ||
+    error{
+        /// Something went wrong while doing math.
+        MathError,
+    };
+
+/// A function that draws a glyph on the provided canvas.
+pub const DrawFn = fn (
+    cp: u32,
+    canvas: *font.sprite.Canvas,
+    width: u32,
+    height: u32,
+    metrics: font.Metrics,
+) DrawFnError!void;
+
+const Range = struct {
+    min: u32,
+    max: u32,
+    draw: DrawFn,
+};
+
+/// Automatically collect ranges for functions with names
+/// in the format `draw<CP>` or `draw<MIN>_<MAX>`.
+const ranges = ranges: {
```

Comment:
Can you add a type signature here? I find with big blocks like this its good practice to prevent future mistyping if we refactor or something. And it just helps self-document a bit

Reply:
Well, it's an array where the size is determined within the block, and you can't do `[_]T` in a type annotation. Would have to move the count code to a separate block to run first if we want an annotation for this.

Reply:
You know what, we can copy it in to a const inside the block and then take the reference of that as a slice, seems to work, I'll do that.